x-pack/legacy/plugins/ml/public/util/app_state_utils.js

```diff
@@ -55,12 +55,14 @@ export function subscribeAppStateToObservable(AppState, appStateName, o$, callba
 
   o$.next(appState[appStateName]);
 
-  o$.pipe(distinctUntilChanged()).subscribe(payload => {
+  const subscription = o$.pipe(distinctUntilChanged()).subscribe(payload => {
```

Comment:
I'm not entirely clear what this line is doing. Could probably do with a comment.

Reply:
Added a comment in 37bfc1c.